apps/seqan_flexbar/general_processing.h

```diff
@@ -483,474 +483,240 @@ void processN(TSeqs& seqs, TIds& ids, TSeqs& seqsRev, TIds& idsRev, TMulti& mult
 }
 
 
-template<typename TSeqs, typename TIds> //Version for single end data
-void preTrim(TSeqs& seqs, TIds& ids, unsigned head, unsigned tail, unsigned min, GeneralStats& stats)
+template<typename TSeqs, typename TIds>
+void _preTrimRemove(const std::vector<TSeqs*> &seqsVector, const std::vector<TIds*> &idsVector, StringSet<bool> const& rem, DemultiplexingParams& demultiplexParams, GeneralStats& stats)
```

Comment:
* Avoid pointers and vectors of pointers.
* Use full template types whenever possible, e.g. TSeqsVector, TIdsVector, ...